pkgs/by-name/br/brainflow/package.nix

```diff
@@ -0,0 +1,86 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchpatch,
+  bluez,
+  cmake,
+  dbus,
+  libftdi1,
+  nix-update-script,
+  pkg-config,
+  useLibFTDI ? true,
+  useOpenMP ? true,
+  buildBluetooth ? true,
+  buildBluetoothLowEnergy ? true,
+  buildONNX ? true,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "brainflow";
+  version = "5.12.1";
+
+  src = fetchFromGitHub {
+    owner = "brainflow-dev";
+    repo = "brainflow";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-haQO03nkvLoXtFVe+C+yi+MwM0CFh6rLcLvU8fQ4k/w=";
+  };
+
+  patches = [
+    # All of these are PRs that were merged into the upstream repository and will be included in the next release
+    # These should be removed once the next version is released
+    (fetchpatch {
+      # Fixes a major issue that prevented the build from working at all (why was this not backported???)
+      url = "https://github.com/brainflow-dev/brainflow/commit/883b0cd08acb99d7b6e241e92fba2e9a363d17b1.patch";
+      hash = "sha256-QQd+BI3I65gfaNS/SKLjCoqbCwPCiTh+nh0tJAZM6hQ=";
+    })
+    (fetchpatch {
+      # Bumps the version of a python dependency that had a backwards-incompatible change
+      url = "https://github.com/brainflow-dev/brainflow/commit/ea23a6f0483ce4d6fdd7a82bace865356ee78d7f.patch";
+      hash = "sha256-dvMpxxRrnJQ9ADGagB1JhuoB9SNwn755wbHzW/3ECeo=";
+    })
+    (fetchpatch {
+      # Fixes an incorrect use of an environment variable during the build
+      url = "https://github.com/brainflow-dev/brainflow/commit/053b8c1253b686cbec49ab4adb47c9ee02d3f99a.patch";
+      hash = "sha256-Pfhe1ZvMagfVAGZqeWn1uHXgwlTtkOm+gyWuvC5/Sro=";
+    })
+  ];
+
+  cmakeFlags = with lib; [
```

Comment:
:/

Reply:
Hey @AndersonTorres, it seems like you're frustrated from seeing this pattern too often, but I don't think it's appropriate to use a smiley face as feedback regardless. Especially when seen by newer contributors, this just makes for an uninviting atmosphere, so I'd appreciate it if you could improve on that :slightly_smiling_face: 

In this case I'd recommend something like

> Using `with lib;` is often [not recommended](https://nix.dev/guides/best-practices#with-scopes), see also https://github.com/NixOS/nixpkgs/pull/330035. I don't think it's necessary here.
> ```suggestion
>   cmakeFlags = [
> ```

If you don't want to type this all the time, you could also (in increasing order of difficulty):
- Use [saved replies](https://docs.github.com/en/get-started/writing-on-github/working-with-saved-replies/using-saved-replies) to not have to type the same response all the time
- Make a PR to document best practices regarding `with` in Nixpkgs, which both solidifies it as a convention for others to propagate, but also allows you to link to it
- Help creating automation to enforce such conventions automatically
- Fix the underlying problem, e.g. by pushing https://github.com/NixOS/rfcs/pull/110